src/env.h

```diff
@@ -991,13 +991,17 @@ struct EnvSerializeInfo {
 };
 
 struct SnapshotData {
-  // The result of v8::SnapshotCreator::CreateBlob() during the snapshot
-  // building process.
-  v8::StartupData v8_snapshot_blob_data;
+  enum class DataOwnership { kOwned, kNotOwned };
 
   static const size_t kNodeBaseContextIndex = 0;
   static const size_t kNodeMainContextIndex = kNodeBaseContextIndex + 1;
 
+  DataOwnership data_ownership;
+
+  // The result of v8::SnapshotCreator::CreateBlob() during the snapshot
+  // building process.
+  v8::StartupData v8_snapshot_blob_data;
```

Comment:
What do you think about
```suggestion
  DataOwnership data_ownership = DataOwnership::kOwned;

  // The result of v8::SnapshotCreator::CreateBlob() during the snapshot
  // building process.
  v8::StartupData v8_startup_data{nullptr, 0};
```
and removing `New()` and the constructor parts? I think this should simplify the class a bit.
This won't add a non-default constructor, so using initializer lists should work too.

Reply:
Thanks, I have added the default initializers and removed the `New()` method.